src/transpiling/jsx_string.rs

```diff
@@ -0,0 +1,1790 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use swc_common::DUMMY_SP;
+use swc_ecma_ast::*;
+use swc_ecma_utils::prepend_stmt;
+use swc_ecma_utils::quote_ident;
+use swc_ecma_visit::{noop_visit_mut_type, VisitMut, VisitMutWith};
+
+struct JsxString {
+  next_index: usize,
+  templates: Vec<(usize, Vec<String>)>,
+  development: bool,
+  import_source: String,
+  import_jsx: Option<Ident>,
+  import_jsx_ssr: Option<Ident>,
+  import_jsx_attr: Option<Ident>,
+  skip_child_serialization: Option<Vec<String>>,
+}
```

Comment:
Let's add `JsxString::new` that will allow to configure various options on creation - `templates` or `next_index` are internal fields that callers shouldn't need to care about.

Reply:
It would also be good to test various options in the test cases